test/tpu/run_tests.sh

```diff
@@ -53,6 +51,8 @@ if [[ -n "$TPU_VERSION" && "$TPU_VERSION" == "4" ]]; then
     python3 examples/eager/train_decoder_only_eager_spmd_data_parallel.py
     python3 examples/eager/train_decoder_only_eager_with_compile.py
     python3 examples/eager/train_decoder_only_eager_multi_process.py
+    python3 test/test_pallas.py
```

Comment:
Is there a reason for moving pallas tests to v4 only?

Reply:
Synced offline. No particular reason. Will move back in a separate PR to unblock pin update.